pkg/controller/disruption/disruption.go

```diff
@@ -433,12 +433,7 @@ func (dc *DisruptionController) getPodsForPdb(pdb *policy.PodDisruptionBudget) (
 	if err != nil {
 		return []*v1.Pod{}, err
 	}
-	// TODO: Do we need to copy here?
-	result := make([]*v1.Pod, 0, len(pods))
-	for i := range pods {
-		result = append(result, &(*pods[i]))
-	}
-	return result, nil
+	return pods, nil
```

Comment:
Did you do audit of where this is called and that any caller is not modifying the received objects?
Also, I would like to see a comment that callers of this method should not modify the received objects.